tilings/algorithms/requirement_placement.py

```diff
@@ -377,15 +377,19 @@ def place_point_of_req(
                 if include_not:
                     res.append(self._tiling.__class__(obs + rem_req, reqs, ass))
                 continue
-
             forced_obs = self.forced_obstructions_from_requirement(
                 gps, indices, cell, direction
             )
+            forced_obs = [
+                o1
+                for o1 in forced_obs
+                if not any(o2 in o1 for o2 in filterfalse(o1.__eq__, forced_obs))
+            ]
             reduced_obs = [o1 for o1 in obs if not any(o2 in o1 for o2 in forced_obs)]
-
+            reduced_obs.extend(filterfalse(reduced_obs.__contains__, forced_obs))
```

Comment:
So we have [minimal forced obs] + [things in obs that don't contain a forced ob]. Is there a chance that something in "obs" *is contained in* something in "forced obs"? If so, this isn't accounted for.

Also, is this line equivalent to
`reduced_obs.extend(fob for fob in forced_obs if fob not in reduced obs)`
? Would it be better to just extend it with all of forced_obs, then cast it to a set to remove duplicates?

Reply:
forced obs come from a requirement, so this is why we do it here, rather than just in the tiling initialiser.

Reply:
In my test, using a set was slower